pkg/version/version_unix.go

```diff
@@ -18,23 +18,31 @@ import (
 
 func parseKernelVersion(ver string) (semver.Version, error) {
 	verStrs := strings.Split(ver, ".")
-	switch {
-	case len(verStrs) < 2:
+
+	if len(verStrs) < 1 {
 		return semver.Version{}, fmt.Errorf("unable to get kernel version from %q", ver)
-	case len(verStrs) < 3:
-		verStrs = append(verStrs, "0")
 	}
-	// We are assuming the kernel version will be something as:
-	// 4.9.17-040917-generic
 
-	// If verStrs is []string{ "4", "9", "17-040917-generic" }
-	// then we need to retrieve patch number.
-	patch := regexp.MustCompilePOSIX(`^[0-9]+`).FindString(verStrs[2])
+	// We are assuming the kernel version will be one of the following:
+	// 4.9.17-040917-generic or 4.9-040917-generic or 4-generic
+
+	// So as observered the kernel value is N.N.N-m or N.N-m or N-m
+	// There for we use regular expresions to extract the patch number
+	// from the last element of the verStrs array and append "0" to
+	// the verStrs array in case the until its lenght is 3 as in all
+	// cases we want to return from this function : Major.Minor.PatchNumber
+
+	patch := regexp.MustCompilePOSIX(`^[0-9]+`).FindString(verStrs[len(verStrs)-1])
 	if patch == "" {
-		verStrs[2] = "0"
+		verStrs[len(verStrs)-1] = "0"
 	} else {
-		verStrs[2] = patch
+		verStrs[len(verStrs)-1] = patch
 	}
+
+	for len(verStrs) <= 3 {
+		verStrs = append(verStrs, "0")
+	}
```

Comment:
If `verStrs` == 3, e.g. `4.9.17`, appending a `0` will cause the version to be `4.9.17.0`. I believe the logic should be:

```
	for len(verStrs) < 3 {
		verStrs = append(verStrs, "0")
	}
```
I suggest adding a test case for this section of code.